core/src/main/java/io/questdb/griffin/engine/join/AsOfJoinNoKeyFastRecordCursorFactory.java

```diff
@@ -121,14 +129,29 @@ public boolean hasNext() {
             if (masterHasNext) {
                 final long masterTimestamp = masterRecord.getTimestamp(masterTimestampIndex);
                 if (masterTimestamp < lookaheadTimestamp) {
+                    if (toleranceInterval != Numbers.LONG_NULL && slaveTimestamp < masterTimestamp - toleranceInterval) {
+                        record.hasSlave(false);
+                    }
                     isMasterHasNextPending = true;
                     return true;
                 }
                 nextSlave(masterTimestamp);
+                if (toleranceInterval != Numbers.LONG_NULL && record.hasSlave()) {
+                    slaveTimestamp = slaveRecB.getTimestamp(slaveTimestampIndex);
+                    if (slaveTimestamp < masterTimestamp - toleranceInterval) {
+                        record.hasSlave(false);
```

Comment:
nit: the branch can be built-in
```suggestion
                        record.hasSlave(slaveTimestamp >= masterTimestamp - toleranceInterval);
```